js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSEnv.scala

```diff
@@ -9,11 +9,47 @@
 
 package org.scalajs.jsenv
 
-import org.scalajs.core.tools.io._
-import org.scalajs.core.tools.classpath._
-import org.scalajs.core.tools.logging._
+import org.scalajs.core.tools.io.VirtualJSFile
+import org.scalajs.core.tools.jsdep.ResolvedJSDependency
+import org.scalajs.core.tools.linker.LinkingUnit
 
 trait AsyncJSEnv extends JSEnv {
-  def asyncRunner(classpath: CompleteClasspath, code: VirtualJSFile,
-      logger: Logger, console: JSConsole): AsyncJSRunner
+  def asyncRunner(libs: Seq[ResolvedJSDependency], sjsCode: VirtualJSFile,
+      code: VirtualJSFile): AsyncJSRunner
+
+  override def loadLibs(libs: Seq[ResolvedJSDependency],
+      sjsCode: VirtualJSFile): AsyncLibJSEnv = {
+    new AsyncLibJSEnv {
+      def unloadLibs: AsyncJSEnv = AsyncJSEnv.this
+
+      def jsRunner(code: VirtualJSFile): JSRunner =
+        AsyncJSEnv.this.jsRunner(libs, sjsCode, code)
+
+      def asyncRunner(code: VirtualJSFile): AsyncJSRunner =
+        AsyncJSEnv.this.asyncRunner(libs, sjsCode, code)
+    }
+  }
+}
+
+trait LinkingUnitAsyncJSEnv extends LinkingUnitJSEnv with AsyncJSEnv {
+  def asyncRunner(libs: Seq[ResolvedJSDependency], linkingUnit: LinkingUnit,
+      code: VirtualJSFile): AsyncJSRunner
```

Comment:
Hum ... I guess Rhino is tightly coupled to the back-end being used, and would crash if we're using a different back-end.

The logical thing to do would be for `LinkingUnitJSEnv` to dictate the back-end to use, or, more specifically, for it to specify its `SymbolRequirement`. When linking with a front-end with the purpose of feeding the linking unit to a `LinkingUnitJSEnv`, we should use that `SymbolRequirement`.

That means `GenLinker.linkUnit` should take a `SymbolRequirement` parameter, rather than using that of its back-end.